pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3017,6 +3026,82 @@ private void internalExpireMessagesForSinglePartition(String subName, int expire
         }
     }
 
+    protected void internalExpireMessagesOnPosition(AsyncResponse asyncResponse, String subName, boolean authoritative,
+                                                 MessageIdImpl messageId, boolean isExcluded, int batchIndex) {
+        if (topicName.isGlobal()) {
+            try {
+                validateGlobalNamespaceOwnership(namespaceName);
+            } catch (Exception e) {
+                log.warn("[{}][{}] Failed to expire messages on subscription {} to position {}: {}", clientAppId(),
+                        topicName, subName, messageId, e.getMessage());
+                resumeAsyncResponseExceptionally(asyncResponse, e);
+                return;
+            }
+        }
+
+        log.info("[{}][{}] received expire messages on subscription {} to position {}", clientAppId(), topicName,
+                subName, messageId);
+
+        // If the topic name is a partition name, no need to get partition topic metadata again
+        if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName, authoritative, false).partitions > 0) {
+            log.warn("[{}] Not supported operation expire message up to {} on partitioned-topic {} {}",
+                    clientAppId(), messageId, topicName, subName);
+            asyncResponse.resume(new RestException(Status.METHOD_NOT_ALLOWED,
+                    "Expire message at position is not supported for partitioned-topic"));
+            return;
+        } else {
+            validateAdminAccessForSubscriber(subName, authoritative);
+            validateReadOperationOnTopic(authoritative);
+            PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
```

Comment:
It need to use `validateTopicOwnership` to check the ownership of the topic. If the owner is not the current request broker, it will return 307 to the client and the client will redirect to the owner broker. Please also add test for cover this case.

I noticed, expire by time also does not handle this. Please help double confirm. It's better to fix it in a separate PR so that we can cherry-pick the fix into the minor release.

Reply:
Actually both `validateAdminAccessForSubscriber` and `validateReadOperationOnTopic` calls `validateTopicOwnership` internally, so should be fine here.

Reply:
https://github.com/apache/pulsar/blob/bd247a982765c9a099f3e15a73b0eee3db17fa71/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L252-L262